### PR TITLE
reduce mem alloc while do etl write

### PR DESCRIPTION
**Pull request by xzxiong**
## What type of PR is this?

- [ ] API-change
- [x] BUG
- [ ] Improvement
- [ ] Documentation
- [ ] Feature
- [ ] Test and CI
- [ ] Code Refactoring

## Which issue(s) this PR fixes:

issue #https://github.com/matrixorigin/matrixone/issues/9273

## What this PR does / why we need it:

memory allocation is too frequent while do etl wirte
modification:
1. change etl file format from `tae` to `csv`, which less memory cost and less cpu cost
2. add struct `ColumnField` and implement `func (r *Row) SetColumnVal(col Column, cf ColumnField)` to ctrl mem alloc
3. adapt `CSVWriter` and `TAEWriter` with `ColumnField` args.
4. improve `MOZapLog::Size` calculation

- ColumnField
```
type ColumnField struct {
	Type      ColType
	Integer   int64
	String    string
	Bytes     []byte
	Interface interface{}
}
```

- result: `GenBatch` alloc objs count < 3.11 %
image

### CODE REVIEW

File: pkg/config/configuration.go [hunk 1]
```diff
@@ -137,10 +137,10 @@ var (
 	defaultSessionTimeout = 24 * time.Hour
 
 	// defaultLogsExtension default: tae. Support val in [csv, tae]
-	defaultLogsExtension = "tae"
+	defaultLogsExtension = "csv"
```
Comment (xzxiong) [inline pkg/config/configuration.go:140]: @daviszhen ptal, Change default config val

**Review by aptend — APPROVED** [review]

File: pkg/util/export/table/table.go [hunk 2]
```diff
@@ -404,17 +415,102 @@ func (tbl *ViewSingleCondition) String() string {
 	return fmt.Sprintf("`%s` = %q", tbl.Column.Name, tbl.Table)
 }
 
+type ColumnField struct {
```
Comment (xzxiong) [inline pkg/util/export/table/table.go:418]: @gavinyue `ColumnField` definition

File: pkg/util/metric/mometric/metric_collector.go [hunk 3]
```diff
@@ -328,30 +328,30 @@ func (s *mfsetETL) GetBatch(ctx context.Context, buf *bytes.Buffer) table.Export
 		for _, metric := range mf.Metric {
 			// reserved labels
 			row.Reset()
-			row.SetColumnVal(metricNameColumn, mf.GetName())
-			row.SetColumnVal(metricNodeColumn, mf.GetNode())
-			row.SetColumnVal(metricRoleColumn, mf.GetRole())
+			row.SetColumnVal(metricNameColumn, table.StringField(mf.GetName()))
```
Comment (xzxiong) [inline pkg/util/metric/mometric/metric_collector.go:331]: @aptend ptal, adapt new api `row.SetCoumnVal(string, ColumnField)`

File: pkg/util/trace/impl/motrace/report_log.go [hunk 4]
```diff
@@ -49,9 +49,18 @@ func (m *MOZapLog) GetName() string {
...
+// level: 5
+// itemName: 8
+// nodeInfo: 40 /*36+4*/
+// spanInfo: 36+16
+const deltaContentLength = int64(26 + 5 + 8 + 40 + 36 + 16)
+
 // Size 计算近似值
 func (m *MOZapLog) Size() int64 {
-	return int64(unsafe.Sizeof(m) + unsafe.Sizeof(len(m.LoggerName)+len(m.Caller)+len(m.Message)+len(m.Extra)))
+	return int64(unsafe.Sizeof(m)) + int64(len(m.LoggerName)+len(m.Caller)+len(m.Message)+len(m.Extra)+len(m.Stack)) +
```
Comment (xzxiong) [inline pkg/util/trace/impl/motrace/report_log.go:62]: @gavinyue ptal, calcualte size of `MOZapLog` with those column may add while output to file.

File: pkg/util/trace/impl/motrace/schema.go [hunk 5]
```diff
@@ -118,7 +118,7 @@ var (
 	messageCol      = table.TextColumn("message", "log message")
 	extraCol        = table.JsonColumn("extra", "log dynamic fields")
 	errCodeCol      = table.StringDefaultColumn("err_code", `0`, "error code info")
-	stackCol        = table.StringWithScale("stack", 4096, "stack info")
+	stackCol        = table.StringWithScale("stack", 2048, "stack info")
```
Comment (xzxiong) [inline pkg/util/trace/impl/motrace/schema.go:121]: shorten the `stack` column

**Review by xzxiong — COMMENTED** [review]
please task a look and review.

**Review by xzxiong — COMMENTED** [review]
please task a look and review.

File: pkg/util/export/batch_processor.go [hunk 6]
```diff
@@ -505,7 +506,7 @@ func (c *MOCollector) Stop(graceful bool) error {
 	c.stopOnce.Do(func() {
 		for len(c.awakeCollect) > 0 && graceful {
 			c.logger.Debug(fmt.Sprintf("doCollect left %d job", len(c.awakeCollect)))
-			time.Sleep(250 * time.Second)
+			time.Sleep(250 * time.Millisecond)
```
Comment (gavinyue) [inline pkg/util/export/batch_processor.go:509]: Why make this shorter?

Comment (xzxiong) [reply pkg/util/export/batch_processor.go:509]: here, waiting c.awakeCollect clean. Normally it will be consumed in short time (like 1 sec)

**Review by gavinyue — APPROVED** [review]

**Review by daviszhen — APPROVED** [review]

**Pull request merged**